src/passes/builtinHandler/blockMethods.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  MemberAccess,
+  Identifier,
+  ExternalReferenceType,
+  Expression,
+  getNodeType,
+} from 'solc-typed-ast';
+import { AST } from '../../ast/ast';
+import { ASTMapper } from '../../ast/mapper';
+import { createCairoFunctionStub, createCallToFunction } from '../../utils/functionGeneration';
+import { createUint256TypeName, createUint8TypeName } from '../../utils/nodeTemplates';
+import { typeNameFromTypeNode } from '../../utils/utils';
+
+/*
+A subpass that replaces the block.timestamp and block.number methods.
+In Solidity these functions return uint256, but in cairo these will return felts.
+Therefore, we wrap the replacement functions in felt_to_uint256 warplib functions.
+*/
+export class BlockMethods extends ASTMapper {
+  visitMemberAccess(node: MemberAccess, ast: AST): void {
+    if (
+      node.vExpression instanceof Identifier &&
+      node.vExpression.name === 'block' &&
+      node.vExpression.vIdentifierType === ExternalReferenceType.Builtin
+    ) {
+      if (node.memberName === 'number') {
+        const replacementCall = createCallToFunction(
+          createCairoFunctionStub(
+            'get_block_number',
+            [],
+            // Chosen since this will be transformed to felt, so width does not matter.
```

Comment:
I find this comment quite misleading. It reads like it's saying that since the result is a felt and we treat uint8s as felts that inherently this is fine. When in actuality it's fine because the call with be wrapped with felt_to_uint256. It's also an important note because of the ongoing PR on structs for numeric types. Might it be a better idea to create a warplib function that returns a uint256 for the block number/timestamp?